File: ruby/hamming/hamming.rb
module BookKeeping
    VERSION = 3
end


class Hamming
	def self.compute(seq1, seq2)
		if seq1.length != seq2.length
			raise ArgumentError
		end
		count = 0
		seq1 = seq1.split("")
		seq2 = seq2.split("")
		(0..seq1.length).each { |index|
			if seq1[index] != seq2[index]
			count += 1
		end
		}
		count
	end
end

File: javascript/hamming/hamming.js
const Hamming = function() {};

Hamming.prototype.compute = function(seq1, seq2){
  //no difference in grabbing seq1 or seq2 length bc they need to be the same
  let len = seq1.length;

  //check for different lengths
  if(len !== seq2.length){
    throw Error('DNA strands must be of equal length.');
  }

  seq1 = seq1.split("");
  seq2 = seq2.split("");

  //count differences between 2 strings
  let count =  0;

  for(var i = 0; i<len;i++){
    if (seq1[i] !== seq2[i]){
      count++;
    }
  }

  return count;

}

module.exports = Hamming;